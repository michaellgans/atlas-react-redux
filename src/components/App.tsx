function App() {
  return <>Hello Atlas</>;
}

export default App;
