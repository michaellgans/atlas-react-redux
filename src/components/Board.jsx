// Board Component

// Asset Imports

// Script Imports

// Returns a Board Component
export function Board() {
    // Define Hook
    return (
    );
}
