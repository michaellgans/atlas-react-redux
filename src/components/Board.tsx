// Board Component

// Asset Imports
import { useAppSelector } from "../store";
import { List } from "./List";
import { RootState } from "../store";

// Returns a Board Component
export function Board() {
  // Define Hook
  const lists = useAppSelector((state: RootState) => state.listsSlice.items);

  return (
    // START BOARD
    <div className="m-auto h-screen w-screen overflow-x-scroll text-center">
      <div className="flex h-full space-x-4">
        {lists.map((list) => (
          <List key={list.id} listID={list.id} listTitle={list.title} />
        ))}
      </div>
    </div>
  );
}
