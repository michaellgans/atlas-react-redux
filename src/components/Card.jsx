// Card Component

// Asset Imports

// Script Imports

// Returns a Card Component
export function Card() {
    // Define Hook
    return (
    );
}
