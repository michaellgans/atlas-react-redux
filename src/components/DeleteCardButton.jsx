// DeleteCardButton Component

// Asset Imports

// Script Imports

// Returns a DeleteCardButton Component
export function DeleteCardButton() {
    // Define Hook
    return (
    );
}
