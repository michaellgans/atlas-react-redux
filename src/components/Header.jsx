// Header Component

// Asset Imports

// Script Imports

// Returns a Header Component
export function Header() {
    // Define Hook
    return (
    );
}
