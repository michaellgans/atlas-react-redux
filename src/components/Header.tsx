// Header Component

// Asset Imports
import logo from "../assets/logo.png";

// Returns a Header Component
export function Header() {
  // Define Hook
  return (
    <header className="sticky top-0 border-b-2 border-blue bg-off-white-light pb-8 pt-8">
      <img className="logo mx-auto w-56" src={logo} alt="logo" />
    </header>
  );
}
