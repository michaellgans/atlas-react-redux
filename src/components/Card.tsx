// Card Component

// Asset Imports
import { DeleteCardButton } from "./DeleteCardButton";

// Script Imports

// Types
type CardProps = {
  title: string;
  description: string;
};

// Returns a Card Component
export function Card({ title, description }: CardProps) {
  // Define Hook
  return (
    <div className="card group/card m-3 flex min-h-24 w-full flex-col items-start rounded bg-off-white-light px-4 py-2 text-blue">
      <h5 className="my-2 flex w-full items-end justify-between text-xl font-black">
        <span>{title}</span>
        {/* DELETE CARD BUTTON */}
        <DeleteCardButton />
      </h5>
      <p className="mt-0 text-left">{description}</p>
    </div>
  );
}
