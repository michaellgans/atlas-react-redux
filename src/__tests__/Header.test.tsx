// Test for Header Component

// Imports
import { render } from "@testing-library/react";
import { expect, test, it } from 'vitest';
import { Header } from "../components/Header";

it('Controll Test', () => {
    expect(2 + 2).toBe(4);
});

test('Header renders correctly', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
});
