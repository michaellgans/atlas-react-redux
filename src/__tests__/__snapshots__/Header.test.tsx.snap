// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="sticky top-0 border-b-2 border-blue bg-off-white-light pb-8 pt-8"
  >
    <img
      alt="logo"
      class="logo mx-auto w-56"
      src="/src/assets/logo.png"
    />
  </header>
</div>
`;
